aten/src/ATen/native/native_functions.yaml

```diff
@@ -3230,6 +3230,31 @@
   variants: function, method
   device_guard: False
 
+- func: hsplit.int(Tensor(a) self, int indices_or_sections) -> Tensor(a)[]
+  use_c10_dispatcher: full
+  variants: function, method
+
+- func: hsplit.array(Tensor(a) self, int[] indices_or_sections) -> Tensor(a)[]
+  use_c10_dispatcher: full
+  variants: function, method
+
+- func: vsplit.int(Tensor(a) self, int indices_or_sections) -> Tensor(a)[]
```

Comment:
Add a third signature corresponding to tensor_split where indices_or_sections can be a tensor